airflow/providers/google/cloud/operators/gcs.py

```diff
@@ -740,19 +740,19 @@ def execute(self, context: "Context") -> List[str]:
         try:
             timespan_start = context["data_interval_start"]
             timespan_end = context["data_interval_end"]
-        except KeyError:  # Data interval context variables are only available in Airflow 2.2+
-            timespan_start = timezone.coerce_datetime(context["execution_date"])
-            timespan_end = timezone.coerce_datetime(context["dag"].following_schedule(timespan_start))
+        except KeyError:
+            timespan_start = context["execution_date"]
```

Comment:
should not we actually check and convert "potential" dt.timetime to pendulum.DateTime here and below ?  that's what corce_datetime did actually.

Reply:
cc: @uranusjr  ?

Reply:
Yes I think we should. Since `context["execution_date"]` can never be None, this can simply use `pendulum.instance`. The `following_schedule()` one below needs an additional check because it _can_ be None.

Reply:
@uranusjr I will convert to the `pendulum` object. but regarding `following_schedule` we checking if the result is `None`, what additional check you have in mind?

Reply:
Simply something like

```python
following_execution_date = context["dag"].following_schedule(context["execution_date"])
if following_execution_date is None:
    timespan_end: Optional[pendulum.DateTime] = None
else:
    timespan_end = pendulum.instance(following_execution_date)
```

Reply:
ok, make sense

Reply:
@potiuk @uranusjr Done